server/src/main/java/org/candlepin/model/PoolCurator.java

```diff
@@ -579,7 +579,8 @@ else if (!addFuture) {
         if (poolIds != null && !poolIds.isEmpty()) {
             criteria = this.currentSession()
                 .createCriteria(Pool.class)
-                .add(CPRestrictions.in("id", poolIds));
+                    .createAlias("product", "Product");
```

Comment:
Almost. :)

The .createAlias should be at the same indentation as the previous line, with the following line as it is now.